src/main/java/org/sunbird/cb/hubservices/serviceimpl/ConnectionService.java

```diff
@@ -30,23 +32,32 @@ public class ConnectionService implements IConnectionService {
 	@Autowired
 	INodeService nodeService;
 	@Override
-	public Response upsert(ConnectionRequest request) {
-		Response response = null;
+	public Response upsert(ConnectionRequest request, String updateOperation) {
+		Response response = new Response();
+		Node from = new Node();
+		Node to = new Node();
 		if(validateRequest(request)) {
-			Node to = new Node(request.getUserIdFrom());
-			Node from = new Node(request.getUserIdTo());
+			if(updateOperation.equalsIgnoreCase(Constants.ADD_OPERATION)) {
+				from = new Node(request.getUserIdFrom());
+				to = new Node(request.getUserIdTo());
+			}
+			else if(updateOperation.equalsIgnoreCase(Constants.UPDATE_OPERATION))
+			{
+				to = new Node(request.getUserIdFrom());
+				from = new Node(request.getUserIdTo());
+			}
 			Map<String, String> relP = setRelationshipProperties(request, from, to);
 			try {
 				nodeService.connect(from, to, relP);
 				if (connectionProperties.isNotificationEnabled())
 					sendNotification(connectionProperties.getNotificationTemplateRequest(), from.getId(),
-							to.getId(), relP.get("status"));
+							to.getId(), relP.get(Constants.STATUS));
 				response.put(Constants.ResponseStatus.MESSAGE, Constants.ResponseStatus.SUCCESSFUL);
```

Comment:
Can we improve this ? check the return value inside nodeService.connect method and make sure the return value is properly returned here... if success then send operation is success otherwise send notification with status as failure...